### ISSUE TITLE
C-g raises "GError, an exception was raised in Javascript"

### ISSUE THREAD

**Issue by vindarel** [atlas-engineer/next#90]
Next is buggy with me :( I start it, I go to the github manual. Here clicking an item of the TOC doesn't work, and invoking `C-g` to jump to links throws

```
GError: Domain: "WebKitPrintError", Code: 699, Message: An exception was raised in JavaScript
   [Condition of type GLIB::G-ERROR-CONDITION]

Restarts:
 0: [ABORT] Abort thread (#<THREAD "cl-cffi-gtk main thread" RUNNING {10071AF623}>)

Backtrace:
  0: (GLIB::MAYBE-RAISE-G-ERROR-CONDITION #.(SB-SYS:INT-SAP #X7FFFD4214CA0))
  1: ((FLET #:CLEANUP-FUN-611 :IN CL-WEBKIT2:WEBKIT-WEB-VIEW-RUN-JAVASCRIPT-FINISH)) [cleanup]
  2: (CL-WEBKIT2:WEBKIT-WEB-VIEW-RUN-JAVASCRIPT-FINISH #<CL-WEBKIT2:WEBKIT-WEB-VIEW {100A4E1523}> #.(SB-SYS:INT-SAP #X7FFFD41D00E0))
  3: ((LAMBDA (INTERFACE::SOURCE-OBJECT INTERFACE::RESULT INTERFACE::USER-DATA) :IN "/home/vince/bacasable/next/next/source/gtk/gtk.lisp") #<unavailable argument> #.(SB-SYS:INT-SAP #X7FFFD41D00E0) #<unavai..
  4: ((LAMBDA (SB-ALIEN::ARGS-POINTER SB-ALIEN::RESULT-POINTER FUNCTION) :IN "/home/vince/.cache/common-lisp/sbcl-1.2.4.debian-linux-x64/home/vince/quicklisp/dists/quicklisp/software/cl-cffi-gtk-20160208-g..
  5: ("foreign function: call_into_lisp")
  6: ("foreign function: funcall3")
  7: ("foreign function: #x201030F5")
  8: ("foreign function: #x201B564408")
  9: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
 10: ((FLET #:WITHOUT-INTERRUPTS-BODY-1226 :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
 11: ((FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
 12: ((FLET #:WITHOUT-INTERRUPTS-BODY-647 :IN SB-THREAD::CALL-WITH-MUTEX))
 13: (SB-THREAD::CALL-WITH-MUTEX #<CLOSURE (FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE) {7FFFDAF3EC8B}> #<SB-THREAD:MUTEX "thread result lock" owner: #<SB-THREAD:THR..
 14: (SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE #<SB-THREAD:THREAD "cl-cffi-gtk main thread" RUNNING {10071AF623}> #S(SB-THREAD:SEMAPHORE :NAME "Thread setup semaphore" :%COUNT 0 :WAITCOUNT 0 :MUTEX #<..
```
The line in cause:

```
(cffi:defcallback js-execution-complete :void ((source-object :pointer)
                                               (result :pointer)
                                               (user-data :pointer))
```

It freezes the window, closing it ("force quit" button) also kills the lisp process.

Debian64 (LMDE),
I have both `libwebkit2gtk-3.0-25` and `libwebkit2gtk-4.0-37` installed.

Is there some files I could delete to try over ?

**Comment by jmercouris** [comment]
Hey, thank you for your report! I'm not sure why that is. We are currently working on a new version which uses a client/server interaction between a Lisp Core and GTK.

This will remove the CFFI interface. What I'm basically saying is, perhaps we don't need to debug it. If you are wanting to have a conversation online, I'd be happy to fill you in on the #next-browser IRC channel on free node,

thanks,

-John

**Comment by vindarel** [comment]
ok, I don't mind waiting a new version. 

regards

**Comment by Ambrevar** [comment]
This should be fixed in Next 1.2.1.  Feel free to re-open if you still experience the issue.
